src/hooks/UseOustsideClick/UseOutsideClick.tsx

```diff
@@ -0,0 +1,23 @@
+'use client';
+
+import { useEffect, useRef } from 'react';
+
+export const useOutsideClick = (callback: () => void) => {
+  const reference = useRef<HTMLUListElement>(null);
```

Comment:
To fix your problem you can create another ref for a button and add it to the condition, also change mousedown to click event

Reply:
Also rename reference to something more specific